### PR TITLE
Add JsonIgnore annotation to avoid debugSupportingFiles creating heap space issues

### PR DESCRIPTION
**Pull request by jlengrand**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by jlengrand** [comment]
Fixes https://github.com/OpenAPITools/openapi-generator/pull/14266. I created a new Pull Request to avoid having any undesired changes from my local copy. Apologies.

This superseeds https://github.com/OpenAPITools/openapi-generator/pull/14266.

**Comment by jlengrand** [comment]
Also, see this comment for context : https://github.com/OpenAPITools/openapi-generator/pull/14266#issuecomment-1353454853

**Comment by wing328** [comment]
Thanks for the PR

cc @OpenAPITools/generator-core-team

**Comment by jlengrand** [comment]
If you have some pointers on how to do a test for this, I'd appreciate it. The only thing is, because it leads to a heap exception, I am not certain what type of test would actually make sense for this.

**Comment by jlengrand** [comment]
Alright, this time pinging @spacether because I am touching the Python implementation. 
As I'm mentioning in the issue, It seems Jackson uses reflection to pick all getters and setters, and because the method is called `getContentTypeToOperation`, it is picked up and generates a crash.

Renaming it to `contentTypeToOperation` fixes the issue.

I propose to rename the method, rather than adding `JsonIgnore` as it seems cleaner.

Do you have strong feeling against it?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenOperation.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 package org.openapitools.codegen;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenOperation.java:20]: @jlengrand thanks for the PR. This line of import is no longer needed after renaming the function below, right?

Comment (jlengrand) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenOperation.java:20]: Very good point! It's a remnant of the old method

**Comment by jlengrand** [comment]
Can I do anything more to get this fixed folks?

**Comment by wing328** [comment]
Tested locally and the fix works for me locally. Will try to add a test to cover it moving forward.

**Comment by wing328** [comment]
CI failure not related to this change.

**Comment by jlengrand** [comment]
> #14331

Awesome, I was unsure how to do this. TY!

**Pull request merged**